src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/ExternalObjectDirectoryStub.java

```diff
@@ -8,15 +8,38 @@
 import uk.gov.hmcts.darts.common.entity.MediaEntity;
 import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
 import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
 
+import java.util.List;
 import java.util.UUID;
 
 @Component
 @RequiredArgsConstructor
 public class ExternalObjectDirectoryStub {
 
     private final UserAccountStub userAccountStub;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalObjectDirectoryRepository eodRepository;
+
+    public ExternalObjectDirectoryEntity createAndSaveEod(MediaEntity media,
+                                                          ObjectRecordStatusEnum objectRecordStatusEnum,
+                                                          ExternalLocationTypeEnum externalLocationTypeEnum) {
+        UUID uuid = UUID.randomUUID();
+        var eod = createExternalObjectDirectory(media, objectRecordStatusEnum, externalLocationTypeEnum, uuid);
+        return eodRepository.save(eod);
+    }
 
+    /**
+     * Creates an ExternalObjectDirectoryEntity.
+     * @deprecated Use
+     *      {@link ExternalObjectDirectoryStub#createExternalObjectDirectory(MediaEntity, ObjectRecordStatusEnum, ExternalLocationTypeEnum, UUID)} instead.
+     */
+    @Deprecated
```

Comment:
Author note:
this will create a bit of noise in the IDE, but I think it's worth it so that we avoid using this method in the future.
We could also gradually replace it with the new one while working on new tickets. 
Replacing the old one with the new is straighforward:

Deprecated:
```
dartsDatabase.getExternalObjectDirectoryStub().createExternalObjectDirectory(
                savedMedia,
                dartsDatabase.getObjectRecordStatusEntity(ARM_DROP_ZONE),
                dartsDatabase.getExternalLocationTypeEntity(ExternalLocationTypeEnum.ARM),
                UUID.randomUUID()
        );
```

New:
```
dartsDatabase.getExternalObjectDirectoryStub().createExternalObjectDirectory(
                savedMedia,
                ARM_DROP_ZONE,
                ARM,
                UUID.randomUUID()
```